### PR TITLE
Add reviews tab to account pages

### PR DESCRIPTION
**Pull request by Pixelguymm**

### CODE REVIEW

File: src/jvmMain/kotlin/io/beatmaps/api/review.kt [hunk 1]
```diff
@@ -152,6 +152,7 @@ fun Route.reviewRoute() {
         val reviews = transaction {
             try {
                 Review
+                    .join(reviewerAlias, JoinType.INNER, Review.userId, reviewerAlias[User.id])
```
Comment (Top-Cat) [inline src/jvmMain/kotlin/io/beatmaps/api/review.kt:155]: Pretty sure this was purposefully omitted. The reviewer is always the user. You should already have that information on the profile page.

File: src/jsMain/kotlin/io/beatmaps/shared/sharedComponents.kt [hunk 2]
```diff
@@ -216,6 +215,31 @@ val playlistOwner = fc<PlaylistOwnerProps> { props ->
     }
 }
 
+external interface ReviewerProps : Props {
```
Comment (Top-Cat) [inline src/jsMain/kotlin/io/beatmaps/shared/sharedComponents.kt:218]: Am I missing the reason this wasn't just a new property on the existing component?

Comment (Pixelguymm) [reply src/jsMain/kotlin/io/beatmaps/shared/sharedComponents.kt:218]: I figured splitting them would be cleaner. You'd never use that property on playlists, or anywhere other than reviews. I can merge them again if you want though, just say the word.

Comment (Top-Cat) [reply src/jsMain/kotlin/io/beatmaps/shared/sharedComponents.kt:218]: It's not a big deal, just results in duplicating the rest of the code. Probably a small enough component to not worry about too much.

File: src/jsMain/kotlin/io/beatmaps/shared/review/main.kt [hunk 3]
```diff
@@ -94,8 +95,10 @@ class ReviewTable : RComponent<ReviewTableProps, ReviewTableState>() {
                 attrs.container = resultsTable
                 attrs.renderElement = InfiniteScrollElementRenderer { rv ->
                     reviewItem {
-                        obj = rv
-                        userId = rv?.creator?.id ?: -1
+                        obj = rv?.apply {
```
Comment (Top-Cat) [inline src/jsMain/kotlin/io/beatmaps/shared/review/main.kt:98]: I think this should probably use copy rather than apply to keep the property a val

**Pull request merged**